index.js

```diff
@@ -1 +1,281 @@
+      source = read('./templates/' + config.testmodule
+        + '/get/get.handlebars', 'utf8');
+    }
+  } else if (operation === 'post') {
+    source = read('./templates/' + config.testmodule
+      + '/post/post.handlebars', 'utf8');
+  } else if (operation === 'put') {
+    source = read('./templates/' + config.testmodule
+      + '/put/put/handlebars', 'utf8');
+  }
+
+  // compile template source and return test string
+  gen = handlebars.compile(source);
+  result = gen(data);
+
+  return result;
+}
+
+/**
+ * Builds a set of unit test stubs for all response codes of a
+ *  path's operation
+ * @param  {json} swagger swagger file containing API
+ * @param  {string} path API path to generate tests for
+ * @param  {string} operation operation of the path to generate tests for
+ * @param  {json} config configuration for testGen
+ * @returns {string|Array} set of all tests for a path's operation
+ */
+function testGenOperation(swagger, path, operation, config) {
+  var responses = swagger.paths[path][operation].responses,
+    result = [], res, test;
+
+  for (res in responses) {
+    if (responses.hasOwnProperty(res)) {
+      result.push(testGenResponse(swagger,
+        path,
+        operation,
+        res,
+        config));
+    }
+  }
+
+  var output = "  describe('" + operation + "', function() {\n";
+
+  for (test in result) {
+    if (result.hasOwnProperty(test)) {
+      output += result[test];
+    }
+  }
+
+  output += '  });\n';
+
+  return output;
+}
+
+/**
+ * Builds a set of unit test stubs for all of a path's operations
+ * @param  {json} swagger swagger file containing API
+ * @param  {string} path API path to generate tests for
+ * @param  {json} config configuration for testGen
+ * @returns {string|Array} set of all tests for a path
+ */
+function testGenPath(swagger, path, config) {
+  var operations = swagger.paths[path],
+    result = [], op;
+
+  for (op in operations) {
+    if (operations.hasOwnProperty(op)) {
+      result.push(testGenOperation(swagger, path, op, config));
+    }
+  }
+
+  var output = "describe('" + path + "', function() {\n", test;
+
+  for (test in result) {
+    if (result.hasOwnProperty(test)) {
+      output += result[test];
+    }
+  }
+
+  output += '});\n';
+
+  return output;
+}
+
+/**
+ * Builds unit test stubs for all paths specified by the configuration
+ * @param  {json} swagger swagger file containing API
+ * @param  {json} config configuration for testGen
+ * @returns {string|Array} set of all tests for a swagger API
+ */
+function testGen(swagger, config) {
+	var paths = swagger.paths,
+    targets = config.pathNames,
+		result = [],
+    output = [],
+    path, ndx, test, i = 0;
+
+  var imports = "'use strict';\n\n"
+    + "var chai = require('chai'),\n"
+    + "  request = require('request'),\n"
+    + "  supertest = require('supertest');\n"
+    + "var api = supertest('" + swagger.schemes[0] + '://' + swagger.host + "'); // supertest init\n\n";
```

Comment:
requiring supertest and using it should be conditional.

And invoke `chai.should()`